functions/mqpublish.go

```diff
@@ -61,14 +61,11 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
 }
 
 func checkin() {
-
-	if err := UpdateHostSettings(); err != nil {
-		logger.Log(0, "failed to update host settings -", err.Error())
-		return
-	}
-
-	if err := PublishGlobalHostUpdate(models.HostMqAction(models.CheckIn)); err != nil {
-		logger.Log(0, "failed to check-in", err.Error())
+	// check/update host settings; publish if changed
+	if UpdateHostSettings() {
+		if err := PublishGlobalHostUpdate(models.HostMqAction(models.CheckIn)); err != nil {
```

Comment:
these are two different host update actions `checkin action` and `update action`, check-in should be sent irrespective of whether there is a change or not
can we just revert these changes to how it was previously?

Reply:
reverted